File: Algorithm_746_Min Cost Climbing Stairs.py
class Solution:
    def minCostClimbingStairs(self, cost):
        """
        :type cost: List[int]
        :rtype: int
        """
        cost.append(0)
        arr=[cost[0],cost[1]]
        for i in range(2,len(cost)): arr.append(min(arr[i-1],arr[i-2])+cost[i])
        return arr[len(arr)-1]

File: Algorithm_746_Min Cost Climbing Stairs.java
class Solution {
    public int minCostClimbingStairs(int[] cost) {
        int n = cost.length;
        for (int i = 2; i < n; i++) {
            cost[i] += Math.min(cost[i - 2], cost[i - 1]);
        }
        return Math.min(cost[n - 2], cost[n - 1]);
    }
}